src/main/infrastructure/service/IABConsentEncoderService.js

```diff
@@ -0,0 +1,47 @@
+import {TCModel, TCString} from '@iabtcf/core'
+import {ConsentEncoderService} from '../../domain/consent/ConsentEncoderService'
+import {BOROS_TCF_ID, BOROS_TCF_VERSION} from '../../core/constants'
+import {inject} from '../../core/ioc/ioc'
+import {GVLFactory} from '../repository/iab/GVLFactory'
+
+class IABConsentEncoderService extends ConsentEncoderService {
+  constructor({gvlFactory = inject(GVLFactory)} = {}) {
+    super()
+    this._gvlFactory = gvlFactory
+  }
+
+  encode({consent = {}, previousEncodedConsent} = {}) {
+    const {vendor = {}, purpose = {}, specialFeatures = {}} = consent
+    const latestGVL = this._gvlFactory.create()
+    const tcModel = previousEncodedConsent
+      ? TCString.decode(previousEncodedConsent)
+      : new TCModel(latestGVL)
+    tcModel.cmpId = BOROS_TCF_ID
+    tcModel.cmpVersion = BOROS_TCF_VERSION
+    tcModel.gvl = latestGVL
```

Comment:
In this case: `TCString.decode(previousEncodedConsent)`
The `tcModel.gvl` should be the latest also?

Reply:
@alextremp I've modified it. Let me know if you see it better now